src/Symfony/Component/Routing/Loader/YamlFileLoader.php

```diff
@@ -127,6 +127,24 @@ protected function parseRoute(RouteCollection $collection, $name, array $config,
         $methods = isset($config['methods']) ? $config['methods'] : array();
         $condition = isset($config['condition']) ? $config['condition'] : null;
 
+        if (isset($requirements['_method'])) {
+            if (0 === count($methods)) {
+                $methods = explode('|', $requirements['_method']);
```

Comment:
@stof An empty array would already be ignored: https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Routing/Route.php#L99-101